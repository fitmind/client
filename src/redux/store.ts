import { Store, createStore, applyMiddleware } from 'redux';
import { ApplicationState, storeInitialState, createRootReducer } from './reducers/root.reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './saga/root.saga';
import { History, createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

// We use hash history because this example is going to be hosted statically.
// Normally you would use browser history.
export const history: History = createBrowserHistory();

export const configureStore = (initialState: ApplicationState): Store<ApplicationState> => {
    const sagaMiddleware = createSagaMiddleware();
    const composeEnhancers = composeWithDevTools({});
    let store = createStore(
        createRootReducer(history),
        initialState,
        composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware)),
    );

    sagaMiddleware.run(rootSaga);
    return store;
};

const store: Store<ApplicationState> = configureStore(storeInitialState);

export default store;
