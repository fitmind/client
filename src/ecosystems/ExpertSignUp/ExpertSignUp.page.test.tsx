import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store, { history } from '../../redux/store';
import { Router } from 'react-router';
import ExpertSignUpPage from './ExpertSignUp.page';

describe('Expert SignUp page', () => {
    it('renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <Router history={history}>
                    <ExpertSignUpPage />
                </Router>
            </Provider>,
        );
    });

    // it('should match the snapshot', () => {
    //     const component = renderer.create(
    //         <Provider store={store}>
    //             <Router history={history}>
    //                 <ExpertSignUpPage />
    //             </Router>
    //         </Provider>,
    //     );
    //     let tree = component.toJSON();
    //     expect(tree).toMatchSnapshot();
    // });
});
