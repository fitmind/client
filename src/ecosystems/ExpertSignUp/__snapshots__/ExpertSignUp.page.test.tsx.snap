// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expert SignUp page should match the snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    />
    <div
      className="col"
    >
      <div
        className="sc-bdVaJa fOsasd"
      >
        <div
          className="card"
        >
          <h5
            className="card-header"
          >
            Expert Registration
          </h5>
          <div
            className="card-body"
          >
            <form
              className=""
              noValidate={true}
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                >
                  First Name
                </label>
                <input
                  className="form-control"
                  name="firstName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="your name"
                  type="text"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                >
                  Last Name
                </label>
                <input
                  className="form-control"
                  name="lastName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="your last name"
                  type="text"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                >
                  Email address
                </label>
                <input
                  className="form-control"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter email"
                  type="email"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="formBasicPassword"
                >
                  Password
                </label>
                <input
                  className="form-control"
                  id="formBasicPassword"
                  name="password"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="password"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
                <small
                  className="text-muted form-text"
                >
                  Password needs to have numbers and letters
                </small>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                  htmlFor="formBasicPassword"
                >
                  Password Confirmation
                </label>
                <input
                  className="form-control"
                  id="formBasicPassword"
                  name="passwordConfirm"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="password"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
                <small
                  className="text-muted form-text"
                >
                  Passwords need to match
                </small>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                >
                  Expertise
                </label>
                <select
                  className="form-control"
                  multiple={true}
                  name="expertise"
                  onBlur={[Function]}
                  onChange={[Function]}
                >
                   
                  <option
                    value="YOGA_COACH"
                  >
                    yoga coach
                  </option>
                  <option
                    value="PERSONAL_COACH"
                  >
                    personal coach
                  </option>
                  <option
                    value="NUTRITIONIST"
                  >
                    nutritionist
                  </option>
                  <option
                    value="LIFE_COACH"
                  >
                    life coach
                  </option>
                </select>
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                >
                  Your Description
                </label>
                <textarea
                  className="form-control"
                  name="description"
                  onBlur={[Function]}
                  onChange={[Function]}
                  rows={4}
                  type="text"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  className="form-label"
                >
                  Phone
                </label>
                <input
                  className="form-control"
                  name="phone"
                  onBlur={[Function]}
                  onChange={[Function]}
                  placeholder="Enter phone"
                  type="text"
                  value=""
                />
                <div
                  className="valid-feedback"
                >
                  Looks good!
                </div>
                <div
                  className="invalid-feedback"
                />
              </div>
              <button
                className="btn btn-outline-primary btn-block"
                disabled={false}
                type="submit"
              >
                Register
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-4"
    />
  </div>
</div>
`;
